pkg/scalers/forgejo_runner_scaler.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+const (
+	defaultForgejoJobsLen = 1
+
+	adminJobMetricPath = "/api/v1/admin/runners/jobs"
+	orgJobMetricPath   = "/api/v1/orgs/%s/actions/runners/jobs"     // /api/v1/orgs/{org}/actions/runners/jobs
+	repoJobMetricPath  = "/api/v1/repos/%s/%s/actions/runners/jobs" // {address}/api/v1/repos/{owner}/{repo}/actions/runners/jobs
+	UserJobMetricPath  = "/api/v1/user/actions/runners/jobs"
+)
+
+type forgejoRunnerMetadata struct {
+	Token      string
+	Address    string
+	MetricPath string
+	Labels     string // comma separated runner labels
+	Global     bool
+	Owner      string
+	Org        string
+	Repo       string
+}
+
+// ForgejoRunnerConfig represents the overall configuration.
+type ForgejoRunnerConfig struct {
+	RunnerMeta forgejoRunnerMetadata `yaml:"runner"` // Runner represents the configuration for the runner.
+}
+
+func parseForgejoRunnerMetadata(config *scalersconfig.ScalerConfig) (*ForgejoRunnerConfig, error) {
+	meta := &ForgejoRunnerConfig{}
+
+	if val, ok := config.TriggerMetadata["token"]; ok && val != "" {
+		meta.RunnerMeta.Token = val
+	} else {
+		return nil, fmt.Errorf("no token given")
+	}
+
+	if val, ok := config.TriggerMetadata["address"]; ok && val != "" {
+		meta.RunnerMeta.Address = val
+	} else {
+		return nil, fmt.Errorf("no address given")
+	}
+
+	if val, ok := config.TriggerMetadata["labels"]; ok && val != "" {
+		meta.RunnerMeta.Labels = val
+	} else {
+		return nil, fmt.Errorf("no labels given")
+	}
+
+	global := false
+	if val, ok := config.TriggerMetadata["global"]; ok && val == stringTrue {
+		global = true
+	}
+
+	meta.RunnerMeta.Global = global
+
+	return meta, nil
+}
+
+type forgejoRunnerScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *ForgejoRunnerConfig
+	client     *http.Client
+	logger     logr.Logger
+}
+
+// NewForgejoRunnerScaler creates a new Forgejo Runner Scaler
+func NewForgejoRunnerScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	c := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)
+
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	meta, err := parseForgejoRunnerMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Forgejo Runner metadata: %w", err)
+	}
+
+	logger := InitializeLogger(config, "forgejo_runner_scaler")
+
+	return &forgejoRunnerScaler{
+		client:     c,
+		metricType: metricType,
+		metadata:   meta,
+		logger:     logger,
+	}, nil
+}
+
+func (s *forgejoRunnerScaler) GetMetricsAndActivity(ctx context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
+	jobList, err := s.getJobsList(ctx)
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, false, err
+	}
+
+	metric := GenerateMetricInMili(metricName, float64(len(jobList.Jobs)))
+
+	metric.Value.Add(resource.Quantity{})
```

Comment:
Why do you need this? I'd say that it's required because `GenerateMetricInMili` returns a formatted metric